drivers/watchdog/wdt_wwdg_stm32.c

```diff
@@ -181,21 +201,22 @@ static int wwdg_stm32_install_timeout(const struct device *dev,
 	WWDG_TypeDef *wwdg = WWDG_STM32_STRUCT(dev);
 	uint32_t timeout = config->window.max * USEC_PER_MSEC;
 	uint32_t calculated_timeout;
-	uint32_t prescaler = 0U;
+	uint32_t prescaler_exp = 0U;
 	uint32_t counter = 0U;
 
 	if (config->callback != NULL) {
 		data->callback = config->callback;
 	}
 
-	wwdg_stm32_convert_timeout(dev, timeout, &prescaler, &counter);
-	calculated_timeout = wwdg_stm32_get_timeout(dev, prescaler, counter);
+	wwdg_stm32_convert_timeout(dev, timeout, &prescaler_exp, &counter);
+	calculated_timeout = wwdg_stm32_get_timeout(dev, prescaler_exp, counter);
 
+	LOG_DBG("prescaler: %d", (1 << prescaler_exp));
 	LOG_DBG("Desired WDT: %d us", timeout);
 	LOG_DBG("Set WDT:     %d us", calculated_timeout);
 
-	if (!(IS_WWDG_PRESCALER(prescaler) && IS_WWDG_COUNTER(counter) &&
-	      IS_WWDG_TIMEOUT(timeout, calculated_timeout))) {
+	if (!(WWDG_COUNTER_IS_VALID(counter) &&
+	      WWDG_TIMEOUT_IS_VALID(timeout, calculated_timeout))) {
```

Comment:
I would prefer we keep existing HAL marco.
I don't see really good reason to remove them, specially if this is to provide alternative implementation using a different name.